.devops/nix/package.nix

```diff
@@ -62,6 +64,28 @@ let
     strings.optionalString (suffices != [ ])
       ", accelerated with ${strings.concatStringsSep ", " suffices}";
 
+  gguf-py = python3.pkgs.buildPythonPackage rec {
+    pname = "gguf";
+    version = "0.0.0";
+    pyproject = true;
+    nativeBuildInputs = with python3.pkgs; [
+      poetry-core
+    ];
+    # buildInputs = [
+    #   python3.pkgs.poetry-core
+    # ];
+    propagatedBuildInputs = with python3.pkgs; [
+      numpy
+    ];
+    src = lib.cleanSource ../../gguf-py;
+    doCheck = false;
+    meta = with lib; {
+      description = "Python package for writing binary files in the GGU format";
+      license = licenses.mit;
+      maintainers = [ maintainers.philiptaron ];
+    };
+  };
```

Comment:
1. You don't have to define this in the let-in, instead you can define `gguf-py` in `scope.nix`. Other packages defined in `scope.nix` can just ask for `gguf-py` as an argument
2. Please make a separate file (e.g. `.../gguf-py/default.nix`) and write it as a function accepting `buildPythonPackage` (`numpy`, `poetry-core`, etc) as an argument so you don't hard-code the python version. You can then `gguf-py = callPackage .../gguf-py { }` from `scope.nix`

Reply:
Cool pattern. I took a dig, can rename/move around the gguf-py package file later.